docs/react/guides/ssr.md

```diff
@@ -1,134 +1,136 @@
+If you've read the Request Waterfalls guide, you might remember this:
 
-### Using `initialData`
+```
+1. |-> Markup (without content)
+2.   |-> JS
+3.     |-> Query
+```
 
-Together with Next.js's [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) or [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering), you can pass the data you fetch in either method to `useQuery`'s' `initialData` option. From React Query's perspective, these integrate in the same way, `getStaticProps` is shown below:
+With a client rendered application, these are the minimum 3 server roundtrips you will need to make before getting any content on the screen for the user. One way of viewing server rendering is that it turns the above into this:
 
-```tsx
-export async function getStaticProps() {
-  const posts = await getPosts()
-  return { props: { posts } }
-}
+```
+1. |-> Markup (with content AND initial data)
+2.   |-> JS
+```
 
-function Posts(props) {
-  const { data } = useQuery({
-    queryKey: ['posts'],
-    queryFn: getPosts,
-    initialData: props.posts,
-  })
+As soon as **1.** is complete, the user can see the content and when **2.** finishes, the page is interactive and clickable. Because the markup also contains the initial data we need, step **3.** does not need to run on the client at all, at least until you want to revalidate the data for some reason.
 
-  // ...
-}
-```
+This is all from the clients perspective. On the server, we need to **prefetch** that data before we generate/render the markup, we need to **dehydrate** that data into a serializable format we can embed in the markup, and on the client we need to **hydrate** that data into a React Query cache so we can avoid doing a new fetch on the client.
 
-The setup is minimal and this can be a quick solution for some cases, but there are a **few tradeoffs to consider** when compared to the full approach:
+Read on to learn how to implement these three steps with React Query.
 
-- If you are calling `useQuery` in a component deeper down in the tree you need to pass the `initialData` down to that point
-- If you are calling `useQuery` with the same query in multiple locations, you need to pass `initialData` to all of them
-- There is no way to know at what time the query was fetched on the server, so `dataUpdatedAt` and determining if the query needs refetching is based on when the page loaded instead
+## A quick note on Suspense
 
-### Using Hydration
+This guide uses the regular `useQuery` API, but as long as you are prefetching your queries you can replace this with `useSuspenseQuery` freely if you wish and things will work the same, but you get to use `<Suspense>` for loading states on the client. In the [advanced guide](../guides/advanced-ssr) we'll switch this around and use `useSuspenseQuery` as the default, as Suspenseful data fetching is necessary for the advanced patterns.
```

Comment:
I don't think you can `useSuspenseQuery` on the server without a framework that understands suspense and streaming. For example, I don't think this works in next12 or even in next13 inside the pages directory (at least it didn't last I checked)

Reply:
I think that as long as the data is already in the cache and fresh so that it doesn't suspend, it'll work? I should probably formulate that clearer though.